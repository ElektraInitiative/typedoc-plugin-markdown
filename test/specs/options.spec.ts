import * as Handlebars from 'handlebars';
import { ProjectReflection } from 'typedoc';

describe(`Options:`, () => {
  describe(`(defaults)`, () => {
    let project: ProjectReflection;
    beforeAll(async () => {
      project = await global.bootstrap(['breadcrumbs.ts'], {});
    });

    test(`should set 'ifShowNamedAnchors' to 'false' by default`, () => {
      expect(
        Handlebars.helpers.ifShowNamedAnchors(global.handlebarsOptionsStub),
      ).toBeFalsy();
    });

    test(`should set 'ifShowBreadcrumbs' to 'true' by default`, () => {
      expect(
        Handlebars.helpers.ifShowBreadcrumbs(global.handlebarsOptionsStub),
      ).toBeTruthy();
    });

    test(`should set 'ifShowPageTitle' to 'true' by default`, () => {
      expect(
        Handlebars.helpers.ifShowPageTitle(global.handlebarsOptionsStub),
      ).toBeTruthy();
    });

    test(`should compile relativeURL helper`, () => {
      const url = (project.getChildByName('Breadcrumbs') as any).url;
      expect(Handlebars.helpers.relativeURL(url)).toMatchSnapshot();
    });
  });

  describe(`(with plugin options)`, () => {
    let project: ProjectReflection;
    beforeAll(async () => {
      project = await global.bootstrap(['breadcrumbs.ts'], {
        publicPath: 'test-public-path',
        namedAnchors: true,
        hideBreadcrumbs: true,
        hidePageTitle: true,
      });
    });

    test(`should set named anchors to 'true'`, () => {
      expect(
        Handlebars.helpers.ifShowNamedAnchors(global.handlebarsOptionsStub),
      ).toBeTruthy();
    });

    test(`should set breadcrumbs to 'false'`, () => {
      expect(
        Handlebars.helpers.ifShowBreadcrumbs(global.handlebarsOptionsStub),
      ).toBeFalsy();
    });

    test(`should set page title to 'false'`, () => {
      expect(
        Handlebars.helpers.ifShowPageTitle(global.handlebarsOptionsStub),
      ).toBeFalsy();
    });

    test(`should compile relativeURL helper with public path`, () => {
      const url = (project.getChildByName('Breadcrumbs') as any).url;
      expect(Handlebars.helpers.relativeURL(url)).toMatchSnapshot();
    });
  });
});
