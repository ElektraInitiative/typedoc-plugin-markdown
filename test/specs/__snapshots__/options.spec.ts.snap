// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options: (defaults) should compile relativeURL helper 1`] = `"Breadcrumbs.md"`;

exports[`Options: (with plugin options) should compile relativeURL helper with public path 1`] = `"test-public-pathclasses/Breadcrumbs.md"`;
